### PR TITLE
Multi-target gRPC packages. Use protobuf directly

### LINKED ISSUE #96: Use Grpc.Net.Client when targeting non-netfx
`Grpc.Core` is out-of-support, replaced with `Grpc.Net.Client`. However, the new library drops netfx support. So, we will need to take on the follow work:

- [ ] Submodule https://github.com/microsoft/durabletask-protobuf directly, as `Microsoft.DurableTask.Sidecar.Protobuf` references `Grpc.Core`.
- [ ] Multi-target `Microsoft.DurableTask.Client.Grpc` and `Microsoft.DurableTask.Worker.Grpc` to `netstandard2.0` and `net6.0` (or net5.0?), and use `Grpc.Core` and `Grpc.Net.Client` under the two targets respectively.

### PR DESCRIPTION
**Pull request by jviau**
This PR does some significant overhauling of gRPC dependencies. The changes breakdown as follows:

1. `Microsoft.DurableTask.Sidecar.Protobuf` dependency removed. https://github.com/microsoft/durabletask-protobuf is not a direct submodule.
1. `Microsoft.DurableTask.Grpc` project & package added, which is purely the dotnet gRPC services for the above protobuf spec.
    - This means it is a duplicate of `Microsoft.DurableTask.Sidecar.Protobuf`, but I figure this belongs in this repo anyways. I am open to a different naming.
    - This does have a meaningful (and necessary) change from the  `Microsoft.DurableTask.Sidecar.Protobuf`: it is multi-target of `netstandard2.0` and `net6.0`. Using `Grpc.Core` and `Grpc.Net.Client` respectively.'
    - I intentionally did not add the `Shared/Grpc` files to this project because I am still unsure what the full picture of this new project is yet. Does it take a dependency on `Microsoft.DurableTask.Abstractions`? Should its name be changed as it is confusing with the two existing gRPC packages?
1. `Microsoft.DurableTask.Client.Grpc` and `Microsoft.DurableTask.Worker.Grpc` have also been multi-targeted, using the newer `Grpc.Net.Client` when targeting `net6.0` and above.
1. Several types were affected to accomplish the multi-targeting. They are minimal changes, mostly branching between `Grpc.Core.Channel` and `Grpc.Net.Client.GrpcChannel` via `#ifdef` statements. 
1. Some gRPC test projects were updated to also be multi-target
    - I couldn't update the integration tests, as they rely on `Microsoft.DurableTask.Sidecar` which is not compatible with netfx.
1. Unnecessary packages were removed from `Microsoft.DurableTask.Worker.Grpc`

Resolves #96

### CODE REVIEW

File: src/Worker/Grpc/Worker.Grpc.csproj [hunk 1]
```diff
@@ -1,7 +1,7 @@
 <Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFramework>netstandard2.0</TargetFramework>
+	  <TargetFrameworks>netstandard2.0;net6.0</TargetFrameworks>
```
Comment (cgillum) [inline src/Worker/Grpc/Worker.Grpc.csproj:4]: Looks like a `\t` snuck in here.

File: test/Worker/Grpc.Tests/DependencyInjection/DurableTaskBuilderExtensionsTests.cs [hunk 2]
```diff
@@ -1,9 +1,17 @@
...
-using Grpc.Core;
 using Microsoft.Extensions.DependencyInjection;
 
+#if NET6_0_OR_GREATER
+using Grpc.Net.Client;
+#endif
+
+#if NETFRAMEWORK
+using Grpc.Core;
+using GrpcChannel = Grpc.Core.Channel;
```
Comment (cgillum) [inline test/Worker/Grpc.Tests/DependencyInjection/DurableTaskBuilderExtensionsTests.cs:12]: Is this something that global usings support? I'm wondering if there is a simple way we can reduce some of this duplication.

File: src/Grpc/Grpc.csproj [hunk 3]
```diff
@@ -0,0 +1,31 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFrameworks>netstandard2.0;net6.0</TargetFrameworks>
+    <PackageDescription>The gRPC Protobuf .NET services for Durable Task Framework.</PackageDescription>
+  </PropertyGroup>
```
Comment (cgillum) [inline src/Grpc/Grpc.csproj:6]: What's the name of this nuget package? I assume it's hidden away via some dynamic MSBuild configuration?

**Review by cgillum — APPROVED** [review]
Yikes, the M1 macOS compatibility issue is unfortunate - I was really hoping we wouldn't need to multitarget like this. 😅

Overall changes look good. However, I'd avoid including `Sidecar` in the package name. `Microsoft.DurableTask.Protobuf` sounds good to me.

Unrelated to this PR, do you think we need a similar change for the Durable extension that runs in the Azure Functions host?

File: src/Grpc/Grpc.csproj (hunk 3 above)
Comment (jviau) [reply src/Grpc/Grpc.csproj:6]: It is `Microsoft.DurableTask.Grpc`

File: test/Worker/Grpc.Tests/DependencyInjection/DurableTaskBuilderExtensionsTests.cs (hunk 2 above)
Comment (jviau) [reply test/Worker/Grpc.Tests/DependencyInjection/DurableTaskBuilderExtensionsTests.cs:12]: Yeah! let me try that.

**Comment by jviau** [comment]
> Unrelated to this PR, do you think we need a similar change for the Durable extension that runs in the Azure Functions host?

Yes, I will look into making a similar change there.

File: src/Grpc/Grpc.csproj (hunk 3 above)
Comment (cgillum) [reply src/Grpc/Grpc.csproj:6]: I'm good with `Microsoft.DurableTask.Grpc` since we have gRPC-specific dependencies we're pulling in.

**Comment by R00iBaard** [comment]
Hey, 

Thanks for all the hard work! Just a question, should the latest release 1.0.0 resolve this issue without manual intervention? Still get the following error on my mac m1

image

**Comment by jviau** [comment]
@R00iBaard we still need to update the Durable Functions host [extension](https://github.com/Azure/azure-functions-durable-extension/tree/dev/src/WebJobs.Extensions.DurableTask). When that is multi-targeted, then it should work without any manual intervention.

**Comment by R00iBaard** [comment]
> @R00iBaard we still need to update the Durable Functions host [extension](https://github.com/Azure/azure-functions-durable-extension/tree/dev/src/WebJobs.Extensions.DurableTask). When that is multi-targeted, then it should work without any manual intervention.

ah great news! Thanks so much

**Pull request merged**